packages/SwingSet/src/supervisors/supervisor-helper.js

```diff
@@ -32,16 +32,15 @@ function makeSupervisorDispatch(dispatch) {
   async function dispatchToVat(delivery) {
     // the (low-level) vat is responsible for giving up agency, but we still
     // protect against exceptions
-    return Promise.resolve(delivery)
-      .then(dispatch)
-      .then(
-        res => harden(['ok', res, null]),
-        err => {
-          // TODO react more thoughtfully, maybe terminate the vat
-          console.warn(`error during vat dispatch() of ${delivery}`, err);
-          return harden(['error', `${err}`, null]);
-        },
-      );
+    await null;
+    try {
+      const res = await dispatch(delivery);
```

Comment:
Technically the previous logic allowed `delivery` to be a promise that would get unwrapped, but I suppose no caller actually did that.

Reply:
Correct, as was already documented with types in the other levels of this stack such as [`deliverAndLogToVat`](https://github.com/Agoric/agoric-sdk/blob/fd22e4a8df77c14efdb448db797773ffdfab4ea2/packages/SwingSet/src/kernel/kernel.js#L427) and [`deliverToVat`](https://github.com/Agoric/agoric-sdk/blob/fd22e4a8df77c14efdb448db797773ffdfab4ea2/packages/SwingSet/src/kernel/vat-warehouse.js#L519) and [`deliver`](https://github.com/Agoric/agoric-sdk/blob/fd22e4a8df77c14efdb448db797773ffdfab4ea2/packages/SwingSet/src/kernel/vat-loader/manager-helper.js#L134). This just brings the stack into closer alignment.